cluster/replication/copier/copier.go

```diff
@@ -363,31 +369,36 @@ func (c *Copier) downloadWorker(ctx context.Context, client FileReplicationServi
 
 			err = wbuf.Flush()
 			if err != nil {
-				return fmt.Errorf("flushing buffer to file %q: %w", localFilePath+".tmp", err)
+				return fmt.Errorf("flushing buffer to file %q: %w", tmpPath, err)
 			}
 
 			err = f.Sync()
 			if err != nil {
-				return fmt.Errorf("fsyncing file %q for writing: %w", localFilePath+".tmp", err)
+				return fmt.Errorf("fsyncing file %q for writing: %w", tmpPath, err)
+			}
+
+			if err := f.Close(); err != nil {
+				return fmt.Errorf("closing file: %w", err)
```

Comment:
in case of return error here, wouldn't the defer logic still try to close the file for the second time?